src/v0/util/index.js

```diff
@@ -1014,12 +1019,19 @@ const getDestinationExternalIDInfoForRetl = (message, destination) => {
 };
 
 const getDestinationExternalIDObjectForRetl = (message, destination) => {
+  const { externalId } = message.context || {};
   let externalIdArray = [];
-  if (message.context && message.context.externalId) {
-    externalIdArray = message.context.externalId;
+
+  if (externalId) {
+    if (Array.isArray(externalId)) {
+      externalIdArray = externalId;
+    } else if (isObject(externalId) && !isEmptyObject(externalId)) {
+      externalIdArray = [externalId];
+    }
   }
+
   let obj;
-  if (externalIdArray) {
+  if (externalIdArray && externalIdArray.length > 0) {
```

Comment:
```suggestion
  if (externalIdArray.length > 0) {
```
We have already defined externalIdArray previously.